common/unsafe/src/test/java/org/apache/spark/unsafe/types/UTF8StringSuite.java

```diff
@@ -515,7 +518,8 @@ public void writeToOutputStreamUnderflow() throws IOException {
     final byte[] test = "01234567".getBytes(StandardCharsets.UTF_8);
 
     for (int i = 1; i <= Platform.BYTE_ARRAY_OFFSET; ++i) {
-      UTF8String.fromAddress(test, Platform.BYTE_ARRAY_OFFSET - i, test.length + i)
+      new UTF8String(
+        new ByteArrayMemoryBlock(test, Platform.BYTE_ARRAY_OFFSET - i, test.length + i))
```

Comment:
I'm surprised this works...

Reply:
I thought this is what you said [here](https://github.com/apache/spark/pull/19222#discussion_r176986304).

Reply:
Looks a bit strange.

Reply:
What is your suggestion to make it non-strange?

Reply:
It depends on what we are testing at this unit test. From the comment in this test `// offset underflow is apparently supported?`, looks like this behavior is uncertain.

Reply:
This change follows the original behavior for the backward compatibility.

If we want to change the behavior, update test, or delete this, it would be good to address in another PR. Is it better to discuss this UT in another PR?

Reply:
Sure, I just say it looks strange. I know you didn't change the unit test's behavior.